ContactAlphabeticalSearchUsingMap.cls

```diff
@@ -0,0 +1,44 @@
+public with sharing class ContactAlphabeticalSearchUsingMap {
+    public List<String> alpabetList{get;set;}
+    public String searchAlphabet{get;set;}
+    public list<contact> contactList{set;get;}
+    public list<contact> showContacts{set;get;}
+    public static contact contacts {set; get;}
+    public map<String,List<Contact>> mapAlphabetRecord{get;set;}
+    public String convertedChar;
+    public String query;
+    public ContactAlphabeticalSearchUsingMap()
+    {
+        mapAlphabetRecord = new map<String,List<Contact>>(); 
+        contactList=new List<Contact>([Select name,email,birthdate,mobilePhone from contact order by Name limit 1000]);
```

Comment:
@Eternus-noopur if you are not sure how many records it will return then always upper limit provided by salesforce. This is ok but when you use uppermost limit then it will serve more records.